Implement basic news feed

dvrylc opened the pull request:
Closes #49 

Includes:
- Basic news feed functionality
- Tests for new models

dvrylc committed:
Implement 3-column UI, closes #48

- Shift content block to center column
- Add placeholder right column for news feed
- Refactor basic styles

dvrylc committed:
Merge remote-tracking branch 'upstream/master' into separate-content-block

dvrylc committed:
Update views for Eatery refactor

dvrylc committed:
Add category placeholder label

dvrylc committed:
Implement basic news feed

Includes:
- Basic news feed functionality
- Tests for new models

dvrylc committed:
Merge remote-tracking branch 'upstream/master' into basic-news-feed

ROHITREDDYBALAM commented on src/main/java/seedu/address/MainApp.java:
```diff
@@ -76,21 +82,36 @@ public void init() throws Exception {
+            if (!feedListOptional.isPresent()) {
+                logger.info("Feed list file not found. Will be starting with an empty FeedList");
+            }
+            initialFeedList = feedListOptional.orElseGet(SampleDataUtil::getSampleFeedList);
```
Better to put some website in the sample feed list so that users can know how the feature works?

dvrylc replied:
There are sample feeds in `SampleDataUtil#getSampleFeedList`

ROHITREDDYBALAM commented on src/main/java/seedu/address/logic/Logic.java:
```diff
@@ -16,10 +18,11 @@
      * @return the result of the command execution.
      * @throws CommandException If an error occurs during command execution.
-     * @throws ParseException If an error occurs during parsing.
+     * @throws ParseException   If an error occurs during parsing.
```
extra space

ROHITREDDYBALAM commented on src/main/java/seedu/address/MainApp.java:
```diff
@@ -76,21 +82,36 @@ public void init() throws Exception {
+
             initialData = new AddressBook();
+            initialFeedList = new FeedList();
         } catch (IOException e) {
```
Change initialData variable to initialAddressBookData since we have feedlist data and might also have other features later?

ROHITREDDYBALAM commented on src/main/java/seedu/address/model/feed/FeedPost.java:
```diff
@@ -0,0 +1,68 @@
+    private final String source;
+    private final String title;
+    private final String address;
+
```
Some people might be confused as to what each of these variables represent. Maybe can consider renaming them?

dvrylc replied:
Any suggestions? I think the current names are pretty intuitive 🤔

ROHITREDDYBALAM commented on src/main/java/seedu/address/model/feed/Feed.java:
```diff
@@ -0,0 +1,118 @@
+                if (buffer1.contains("<title>")) {
+                    buffer2 = in.readLine();
+
+                    if (buffer1.contains("<title>") && buffer2.contains("<link>")) {
```
checking twice if buffer1 contains title?

ROHITREDDYBALAM commented on src/main/java/seedu/address/model/feed/UniqueFeedList.java:
```diff
@@ -0,0 +1,132 @@
+            throw new FeedNotFoundException();
+        }
+
+        if (!target.equals(editedFeed) && contains(editedFeed)) {
+            throw new DuplicateFeedException();
+        }
```
What happens if target feed equals edited feed?

dvrylc replied:
If target equals edited, it's safe to perform a `setFeed` since there won't actually be any effective change so we don't throw an exception

ROHITREDDYBALAM commented on src/main/java/seedu/address/model/feed/UniqueFeedList.java:
```diff
@@ -0,0 +1,132 @@
+     */
+    public void setFeeds(List<Feed> feeds) {
+        requireAllNonNull(feeds);
+        if (!feedsAreUnique(feeds)) {
+            throw new DuplicateFeedException();
+        }
```
Even if one feed in the list is duplicated, it will throw an exception and nothing is done with the other feeds in the list..? Instead can remove the duplicate and add the others?

dvrylc replied:
The aim of `setFeeds` is to replace the entire collection of feeds with a new set. If the new set contains duplicates, it's not a valid `UniqueFeedList` and so we shouldn't do anything to the existing list.

dvrylc committed:
Improve code style

- Fix indentation
- Improve variable names
- Remove unnecessary `if` conditions

ROHITREDDYBALAM reviewed (approved):
LGTM! Should remember to add test cases later on
